packages/modules/manage-versions/src/components/ManageVersions/ManageVersions.tsx

```diff
@@ -253,12 +301,26 @@ export const ManageVersions = (props: ManageVersionsProps) => {
         skip: changesets?.length,
       })
       .then((newChangesets) => {
-        setChangesetStatus(RequestStatus.Finished);
         setChangesets([...(changesets ?? []), ...newChangesets]);
         updateChangesetsProperties(newChangesets);
+        setChangesetStatus(RequestStatus.Finished);
       })
       .catch(() => setChangesetStatus(RequestStatus.Failed));
-  }, [changesetClient, changesets, imodelId]);
+  }, [changesets, changesetClient, imodelId]);
+
+  React.useEffect(() => {
+    if (versionStatus === RequestStatus.NotStarted) {
+      getVersions();
+      getChangesets();
+    }
+  }, [getChangesets, getVersions, versionStatus]);
+
+  React.useEffect(() => {
+    if (shouldPerformVersionMapping.current) {
```

Comment:
this effect will not be triggered on ref change.
Do you know that? How it is expected to work here...

Reply:
It will trigger as ref is added in dependency array of useEffect.
I need this mutable ref because calling mapChangesetsToNamedVersions() directly does not work as it has dependencies on versions and changesets both and loading of nv and changesets are async

Reply:
Removed ref used before, and load changeset on expand in version table